<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>758</width>
    <height>479</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Simple Image</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resource.qrc">
    <normaloff>:/icon/icon/filter.png</normaloff>:/icon/icon/filter.png</iconset>
  </property>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>758</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>ファイル(&amp;F)</string>
    </property>
    <addaction name="openAct"/>
    <addaction name="openClipboardAct"/>
    <addaction name="saveAsAct"/>
    <addaction name="separator"/>
    <addaction name="closeAct"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>ヘルプ(&amp;H)</string>
    </property>
    <addaction name="aboutAct"/>
    <addaction name="aboutQtAct"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>編集(&amp;E)</string>
    </property>
    <addaction name="undoAct"/>
    <addaction name="redoAct"/>
    <addaction name="separator"/>
    <addaction name="copyAct"/>
    <addaction name="pasteAct"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>表示(&amp;V)</string>
    </property>
    <widget class="QMenu" name="menuToolBar">
     <property name="statusTip">
      <string>ツールバーの表示/非表示を切り替えます。</string>
     </property>
     <property name="title">
      <string>ツールバー(&amp;B)</string>
     </property>
     <addaction name="fileToolBarVisibleAct"/>
     <addaction name="editToolBarVisibleAct"/>
     <addaction name="filterToolBarVisibleAct"/>
     <addaction name="zoomToolBarVisibleAct"/>
    </widget>
    <widget class="QMenu" name="menuZoom">
     <property name="title">
      <string>ズーム(&amp;Z)</string>
     </property>
     <addaction name="zoomFullScaleAct"/>
     <addaction name="autoImageResizeAct"/>
     <addaction name="zoomInAct"/>
     <addaction name="zoomOutAct"/>
    </widget>
    <addaction name="menuZoom"/>
    <addaction name="separator"/>
    <addaction name="menuToolBar"/>
    <addaction name="statusBarVisibleAct"/>
    <addaction name="tabVisibleAct"/>
   </widget>
   <widget class="QMenu" name="menuOperate">
    <property name="title">
     <string>操作(&amp;O)</string>
    </property>
    <widget class="QMenu" name="menuOpeColor">
     <property name="title">
      <string>色(&amp;C)</string>
     </property>
     <addaction name="filterGrayscaleAct"/>
     <addaction name="filterTo1BitAct"/>
    </widget>
    <widget class="QMenu" name="menuOpeFilter">
     <property name="title">
      <string>フィルタ(&amp;F)</string>
     </property>
     <addaction name="linearFilterAct"/>
     <addaction name="medianFilterAct"/>
     <addaction name="normFilterAct"/>
     <addaction name="filterInsertRandomNoiseAct"/>
    </widget>
    <addaction name="menuOpeColor"/>
    <addaction name="menuOpeFilter"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuOperate"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QToolBar" name="fileToolBar">
   <property name="windowTitle">
    <string>ファイル</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="openAct"/>
   <addaction name="saveAsAct"/>
  </widget>
  <widget class="QToolBar" name="editToolBar">
   <property name="windowTitle">
    <string>編集</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="undoAct"/>
   <addaction name="redoAct"/>
   <addaction name="separator"/>
   <addaction name="copyAct"/>
   <addaction name="pasteAct"/>
  </widget>
  <widget class="QToolBar" name="filterToolBar">
   <property name="windowTitle">
    <string>フィルタ</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="linearFilterAct"/>
   <addaction name="medianFilterAct"/>
   <addaction name="normFilterAct"/>
  </widget>
  <widget class="QToolBar" name="zoomToolBar">
   <property name="windowTitle">
    <string>ズーム</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="zoomInAct"/>
   <addaction name="zoomOutAct"/>
  </widget>
  <action name="openAct">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/open.png</normaloff>:/icon/icon/open.png</iconset>
   </property>
   <property name="text">
    <string>開く(&amp;O)...</string>
   </property>
   <property name="statusTip">
    <string>ファイルを開きます。</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="saveAsAct">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/save.png</normaloff>:/icon/icon/save.png</iconset>
   </property>
   <property name="text">
    <string>名前を付けて保存(&amp;A)...</string>
   </property>
   <property name="toolTip">
    <string>名前を付けて保存(A)</string>
   </property>
   <property name="statusTip">
    <string>ファイルに保存します。</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="closeAct">
   <property name="text">
    <string>閉じる(&amp;X)</string>
   </property>
   <property name="statusTip">
    <string>プログラムを終了します。</string>
   </property>
  </action>
  <action name="aboutAct">
   <property name="text">
    <string>バージョン情報(&amp;A)</string>
   </property>
  </action>
  <action name="copyAct">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/copy.png</normaloff>:/icon/icon/copy.png</iconset>
   </property>
   <property name="text">
    <string>コピー(&amp;C)</string>
   </property>
   <property name="statusTip">
    <string>編集中の画像をクリップボードにコピーします。</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="pasteAct">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/paste.png</normaloff>:/icon/icon/paste.png</iconset>
   </property>
   <property name="text">
    <string>貼り付け(&amp;P)</string>
   </property>
   <property name="statusTip">
    <string>クリップボードの画像を貼り付けます。</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="statusBarVisibleAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ステータスバー(&amp;S)</string>
   </property>
   <property name="statusTip">
    <string>ステータスバーを隠します。</string>
   </property>
  </action>
  <action name="aboutQtAct">
   <property name="text">
    <string>About &amp;Qt</string>
   </property>
  </action>
  <action name="filterGrayscaleAct">
   <property name="text">
    <string>グレースケール化(&amp;G)</string>
   </property>
   <property name="statusTip">
    <string>カラー画像を輝度値からグレースケールにします。</string>
   </property>
  </action>
  <action name="undoAct">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/undo.png</normaloff>:/icon/icon/undo.png</iconset>
   </property>
   <property name="text">
    <string>元に戻す(&amp;Z)</string>
   </property>
   <property name="statusTip">
    <string>前の操作に戻ります。</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="redoAct">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/redo.png</normaloff>:/icon/icon/redo.png</iconset>
   </property>
   <property name="text">
    <string>やり直し(&amp;Y)</string>
   </property>
   <property name="statusTip">
    <string>元に戻した操作をやり直します。</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="filterTo1BitAct">
   <property name="text">
    <string>2値化(&amp;B)...</string>
   </property>
   <property name="statusTip">
    <string>多値画像を白と黒の2値画像にします。</string>
   </property>
  </action>
  <action name="linearFilterAct">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/filter.png</normaloff>:/icon/icon/filter.png</iconset>
   </property>
   <property name="text">
    <string>線形フィルタ(&amp;L)...</string>
   </property>
   <property name="statusTip">
    <string>近傍画素に係数をかけた総和を画素値にします。</string>
   </property>
  </action>
  <action name="medianFilterAct">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/median.png</normaloff>:/icon/icon/median.png</iconset>
   </property>
   <property name="text">
    <string>メディアンフィルタ(&amp;M)</string>
   </property>
   <property name="statusTip">
    <string>近傍 3x3 画素の中央値を新たな画素値にします。</string>
   </property>
  </action>
  <action name="fileToolBarVisibleAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ファイル(&amp;F)</string>
   </property>
   <property name="statusTip">
    <string/>
   </property>
  </action>
  <action name="normFilterAct">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/norm.png</normaloff>:/icon/icon/norm.png</iconset>
   </property>
   <property name="text">
    <string>勾配ノルムフィルタ(&amp;G)...</string>
   </property>
   <property name="statusTip">
    <string>勾配 (gradient) のノルムを計算します。</string>
   </property>
  </action>
  <action name="filterToolBarVisibleAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>フィルタ(&amp;T)</string>
   </property>
  </action>
  <action name="editToolBarVisibleAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>編集(&amp;E)</string>
   </property>
  </action>
  <action name="openClipboardAct">
   <property name="text">
    <string>クリップボードから開く(&amp;C)</string>
   </property>
   <property name="statusTip">
    <string>クリップボードの画像を開きます。</string>
   </property>
  </action>
  <action name="tabVisibleAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>タブ(&amp;T)</string>
   </property>
   <property name="statusTip">
    <string>画像をタブで選べるようにします。</string>
   </property>
  </action>
  <action name="zoomToolBarVisibleAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ズーム(&amp;Z)</string>
   </property>
  </action>
  <action name="zoomInAct">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/zoom_in.png</normaloff>:/icon/icon/zoom_in.png</iconset>
   </property>
   <property name="text">
    <string>ズームイン(&amp;I)</string>
   </property>
   <property name="statusTip">
    <string>拡大表示します。</string>
   </property>
   <property name="shortcut">
    <string>+</string>
   </property>
  </action>
  <action name="zoomOutAct">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icon/icon/zoom_out.png</normaloff>:/icon/icon/zoom_out.png</iconset>
   </property>
   <property name="text">
    <string>ズームアウト(&amp;O)</string>
   </property>
   <property name="statusTip">
    <string>縮小表示します。</string>
   </property>
   <property name="shortcut">
    <string>-</string>
   </property>
  </action>
  <action name="zoomFullScaleAct">
   <property name="text">
    <string>原寸大(&amp;F)</string>
   </property>
   <property name="statusTip">
    <string>画像のサイズで表示します。</string>
   </property>
  </action>
  <action name="autoImageResizeAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>画面に合わせる(&amp;A)</string>
   </property>
   <property name="statusTip">
    <string>画面サイズに合わせて画像サイズを変更します。</string>
   </property>
  </action>
  <action name="filterInsertRandomNoiseAct">
   <property name="text">
    <string>ランダムノイズの追加(R)...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>closeAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aboutAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aboutQtAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>aboutQt()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveAsAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>statusBarVisibleAct</sender>
   <signal>triggered(bool)</signal>
   <receiver>statusBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>288</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>copyAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pasteAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileToolBarVisibleAct</sender>
   <signal>triggered(bool)</signal>
   <receiver>fileToolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileToolBar</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>fileToolBarVisibleAct</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterToolBarVisibleAct</sender>
   <signal>triggered(bool)</signal>
   <receiver>filterToolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>486</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterToolBar</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>filterToolBarVisibleAct</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>486</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editToolBarVisibleAct</sender>
   <signal>triggered(bool)</signal>
   <receiver>editToolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editToolBar</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>editToolBarVisibleAct</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterGrayscaleAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>filterGrayscale()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterTo1BitAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>filterTo1Bit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>linearFilterAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>linearFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>medianFilterAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>medianFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>normFilterAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>normFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openClipboardAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openClipboard()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabVisibleAct</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setTabVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomToolBarVisibleAct</sender>
   <signal>triggered(bool)</signal>
   <receiver>zoomToolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomToolBar</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>zoomToolBarVisibleAct</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>539</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomInAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomOutAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomFullScaleAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>zoomFullScale()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterInsertRandomNoiseAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>filterInsertRandomNoise()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveAs()</slot>
  <slot>copy()</slot>
  <slot>paste()</slot>
  <slot>about()</slot>
  <slot>aboutQt()</slot>
  <slot>updateMenus()</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>filterGrayscale()</slot>
  <slot>filterTo1Bit()</slot>
  <slot>linearFilter()</slot>
  <slot>medianFilter()</slot>
  <slot>normFilter()</slot>
  <slot>openClipboard()</slot>
  <slot>setTabVisible(bool)</slot>
  <slot>open()</slot>
  <slot>zoomIn()</slot>
  <slot>zoomOut()</slot>
  <slot>zoomFullScale()</slot>
  <slot>filterInsertRandomNoise()</slot>
 </slots>
</ui>
